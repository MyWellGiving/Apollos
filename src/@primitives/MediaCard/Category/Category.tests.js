import React from 'react';
import renderer from 'react-test-renderer';

import ThemeProvider from '@primitives/ThemeProvider';
import Category from './';

describe('the MediaCard Category component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <Category type={'Series'} />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
