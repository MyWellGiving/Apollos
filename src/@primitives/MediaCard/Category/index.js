import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes } from 'recompose';

import withTheme from '@primitives/withTheme';
import styled from '@primitives/styled';
import { H7 } from '@primitives/typography';
import Icon from '@primitives/Icon';
import rem from '@utils/remUnit';

const enhance = compose(
  pure,
  withTheme(),
  setPropTypes({
    type: PropTypes.string,
    color: PropTypes.string,
  }),
);

const Wrapper = styled(({ theme }) => ({
  flexDirection: 'row',
  alignItems: 'center',
  flex: 1,
  paddingTop: theme.baseUnit / 2,
  paddingHorizontal: theme.baseUnit,
  paddingBottom: theme.baseUnit,
}))(View);

const StyledH7 = styled(({ theme, fontColor }) => ({
  paddingHorizontal: theme.baseUnit / 2,
  color: fontColor,
}))(H7);

const Category = enhance(({
  type,
  color,
  theme,
}) => (
  <Wrapper>
    <Icon
      name={'video'}
      size={rem(1.2, theme)}
      fill={color}
    />
    <StyledH7 fontColor={color}>{type}</StyledH7>
  </Wrapper>
));

export default Category;
