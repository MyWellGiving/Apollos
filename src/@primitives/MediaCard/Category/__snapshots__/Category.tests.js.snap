// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the MediaCard Category component should render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "paddingBottom": 20,
      "paddingHorizontal": 20,
    }
  }
>
  <Svg
    height={21.6}
    viewBox="0 0 24 24"
    width={21.6}
  >
    <Path
      d="M3.8182 5.7778v12.4444H14.2V5.7778H3.8182zM16 6v12c0 1.1046-.8954 2-2 2H4c-1.1046 0-2-.8954-2-2V6c0-1.1046.8954-2 2-2h10c1.1046 0 2 .8954 2 2zm4.1818 2.6952L16 11.8084v.3832l4.1818 3.251V8.695zM16 14.4957V9.4253l6-4.2178v13.585l-6-4.2968z"
      fill="#505050"
    />
  </Svg>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#505050",
        "fontFamily": null,
        "fontSize": 14,
        "lineHeight": 15.87,
        "paddingHorizontal": 10,
      }
    }
  >
    Series
  </Text>
</View>
`;
